support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -0,0 +1,268 @@
+			margin-bottom: ${space[2]}px;
+		}
+	}
+`;
+
+const searchParams = new URLSearchParams(window.location.search);
+const query = {
+	productId: searchParams.get('product'),
+	ratePlanId: searchParams.get('ratePlan'),
+	chargeId: searchParams.get('charge'),
+};
+
+const countryGroupIds = ['uk', 'us', 'eu', 'au', 'nz', 'ca', 'int'] as const;
+const CountryGroupIdSchema = picklist(countryGroupIds);
+const countryGroupId = parse(
+	CountryGroupIdSchema,
+	window.location.pathname.split('/')[1],
+);
+
+let currentCurrency: Currency;
+let currentCurrencyKey: keyof typeof currencies;
+switch (countryGroupId) {
+	case 'uk':
+		currentCurrency = currencies.GBP;
+		currentCurrencyKey = 'GBP';
+		break;
+	case 'us':
+		currentCurrency = currencies.USD;
+		currentCurrencyKey = 'USD';
+		break;
+	case 'au':
+		currentCurrency = currencies.AUD;
+		currentCurrencyKey = 'AUD';
+		break;
+	case 'eu':
+		currentCurrency = currencies.EUR;
+		currentCurrencyKey = 'EUR';
+		break;
+	case 'nz':
+		currentCurrency = currencies.NZD;
+		currentCurrencyKey = 'NZD';
+		break;
+	case 'ca':
+		currentCurrency = currencies.CAD;
+		currentCurrencyKey = 'CAD';
+		break;
+}
+
+const ProductSchema = object({
+	ratePlans: record(
+		object({
+			id: string(),
+			charges: record(
+				object({
+					id: string(),
+					pricing: record(number()),
+				}),
+			),
+		}),
+	),
+});
+
+const ProductsSchema = object({
+	products: record(ProductSchema),
+});
+type Products = Output<typeof ProductsSchema>;
+
+function describeProduct(
+	productId: string,
+	ratePlanId: string,
+	chargeId: string,
+) {
+	let description = `${productId} - ${ratePlanId} ${chargeId}`;
+	let frequency = '';
+
+	if (productId === 'HomeDelivery') {
+		frequency = 'month';
+		description = `${ratePlanId} paper`;
+
+		if (ratePlanId === 'Sixday') {
+			description = 'Six day paper';
+		}
+		if (ratePlanId === 'Everyday') {
+			description = 'Every day paper';
+		}
+		if (ratePlanId === 'Weekend') {
+			description = 'Weekend paper';
+		}
+		if (ratePlanId === 'Saturday') {
+			description = 'Saturday paper';
+		}
+		if (ratePlanId === 'Sunday') {
+			description = 'Sunday paper';
+		}
+	}
+
+	if (
+		productId === 'GuardianWeeklyDomestic' ||
+		productId === 'GuardianWeeklyRestOfWorld'
+	) {
+		if (ratePlanId === 'OneYearGift') {
+			frequency = 'year';
+			description = 'The Guardian Weekly Gift Subscription';
+		}
+		if (ratePlanId === 'Annual') {
+			frequency = 'year';
+			description = 'The Guardian Weekly';
+		}
+		if (ratePlanId === 'Quarterly') {
+			frequency = 'quarter';
+			description = 'The Guardian Weekly';
+		}
+		if (ratePlanId === 'Monthly') {
+			frequency = 'month';
+			description = 'The Guardian Weekly';
+		}
+		if (ratePlanId === 'ThreeMonthGift') {
+			frequency = 'quarter';
+			description = 'The Guardian Weekly Gift Subscription';
+		}
+		if (ratePlanId === 'SixWeekly') {
+			frequency = 'month';
+			description = 'The Guardian Weekly';
+		}
+	}
+
+	return { description, frequency };
+}
```

Comment:
I did think of having a `SourceProduct (the API def) => RenderingProduct` transformation, but I have often found those transformations confusing, and often copying most data with a little augmentation.

Similar to the `CountryGroupId => T` transformers, I think we should endevour to only have 1 level deep `Product => T` transformers.